lib/internal/crypto/hkdf.js

```diff
@@ -139,40 +138,35 @@ function hkdfSync(hash, key, salt, info, length) {
   return bits;
 }
 
+const hkdfPromise = promisify(hkdf);
 async function hkdfDeriveBits(algorithm, baseKey, length) {
   const { hash } = algorithm;
   const salt = getArrayBufferOrView(algorithm.salt, 'algorithm.salt');
   const info = getArrayBufferOrView(algorithm.info, 'algorithm.info');
   if (hash === undefined)
     throw new ERR_MISSING_OPTION('algorithm.hash');
 
-  let byteLength = 512 / 8;
-  if (length !== undefined) {
-    if (length === 0)
-      throw lazyDOMException('length cannot be zero', 'OperationError');
-    if (length === null)
-      throw lazyDOMException('length cannot be null', 'OperationError');
-    validateUint32(length, 'length');
-    if (length % 8) {
-      throw lazyDOMException(
-        'length must be a multiple of 8',
-        'OperationError');
-    }
-    byteLength = length / 8;
+  if (length === 0)
+    throw lazyDOMException('length cannot be zero', 'OperationError');
+  if (length === null)
+    throw lazyDOMException('length cannot be null', 'OperationError');
+  if (length % 8) {
+    throw lazyDOMException(
+      'length must be a multiple of 8',
+      'OperationError');
   }
 
-  return new Promise((resolve, reject) => {
-    hkdf(
-      normalizeHashName(hash.name),
-      baseKey[kKeyObject],
-      salt,
-      info,
-      byteLength,
-      (err, bits) => {
-        if (err) return reject(err);
-        resolve(bits);
-      });
-  });
+  let bits;
+  try {
+    bits = await hkdfPromise(
```

Comment:
nit: any reason to use a variable here? It seems we could directly return.
```suggestion
  try {
    return await hkdfPromise(
```